packages/rs-drive/src/drive/contract/queries.rs

```diff
@@ -25,7 +25,10 @@ impl Drive {
     /// * `PathQuery` - A `PathQuery` object representing the query for fetching the contract data.
     pub fn fetch_contract_query(contract_id: [u8; 32]) -> PathQuery {
         let contract_path = contract_root_path_vec(contract_id.as_slice());
-        PathQuery::new_single_key(contract_path, vec![0])
+        let mut query = PathQuery::new_single_key(contract_path, vec![0]);
+
+        query.query.limit = Some(1);
+        query
```

Comment:
The problem is that you can't merge PathQueries with limits. It might be best to have a bool for "explicit_limit".

Reply:
`fetch_contract_query` and `fetch_contract_with_history_latest_query` are only used in `verify_contract_v0` and `prove_contract_v0`. 

I would suggest to keep limits here (at least for now) and add some comments.

Reply:
Fixed in another PR